test/Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -147,6 +147,7 @@ public class ResponseTests : LoggedTest
         {
             var onStartingCalled = false;
             var onCompletedCalled = false;
+            var onCompleted = Task.Run(() => onCompletedCalled = true);
```

Comment:
This test modification isn't correct. This task could run immediately and you wouldn't be able to tell if OnCompleted was called or not.